src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as actionTodos } from './features/todos';
+import { SortType } from './types/SortType';
 
 export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const selectTodo = useAppSelector((state) => state.currentTodo);
+  const todos = useAppSelector((state) => state.todos);
+  const filterState = useAppSelector((state) => state.filter);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos();
+
+        dispatch(actionTodos.SetAdd(loadedTodos));
+        setIsLoading(false);
+      } catch (error) {
+        if (error instanceof Error) {
+          setErrorMessage(error.message);
+        }
+
+        setIsLoading(false);
```

Comment:
`setIsLoading(false); ` repeatable code can be moved to the [`finally` block](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)